src/subdomains/core/history/controllers/transaction.controller.ts

```diff
@@ -80,6 +83,27 @@ export class TransactionController {
     return this.getHistoryData(query, ExportType.COMPACT, res);
   }
 
+  @Get('uid/:uid')
+  @ApiOkResponse({ type: TransactionDto })
+  async getSingleTransaction(@Param('uid') uid: string): Promise<TransactionDto> {
+    const transaction = await this.transactionService.getTransactionByUid(uid, { buyCrypto: true, buyFiat: true });
+    if (!transaction) throw new NotFoundException('Transaction not found');
+    if (!transaction.txTarget) throw new BadRequestException('Unsupported transaction type');
+
+    switch (transaction.txTarget.constructor) {
```

Comment:
Remove the check and change the switch to
`switch (transaction.txTarget?.constructor) {`